File: day2/part2.cpp
#include <iostream>
#include <fstream>

int main()
{
	std::ifstream input{"input.txt"};

	std::string direction;
	int distance;

	int horizontalPosition = 0;
	int aim = 0;
	int depth = 0;

	std::cout << std::boolalpha << true << false << '\n';

	while (input >> direction >> distance) {
		if (direction == "forward") {
			horizontalPosition += distance;
			depth += aim * distance;
		} else if (direction == "down") {
			aim += distance;
		} else if (direction == "up") {
			aim -= distance;
		}
	}

	std::cout << horizontalPosition * depth << std::endl;
}

File: day2/part2.py
import time
start = time.process_time()

for i in range(10000):
	for instruction, amount in map(str.split, open("input.txt")):
		aim = depth = horizontal_pos = 0
		amount = int(amount)
		if instruction == "down":
			aim += amount
		elif instruction == "up":
			aim  -= amount
		elif instruction == "forward":
			horizontal_pos += amount
			depth += aim * amount

print(time.process_time() - start)  # 5.6
print(f"Submarine @ {horizontal_pos, depth}")
print("Answer is", horizontal_pos * depth)